code/game/mecha/mecha_actions.dm

```diff
@@ -0,0 +1,281 @@
+//////////////////////////////////////// Action Buttons ///////////////////////////////////////////////
+
+/obj/mecha/proc/GrantActions(mob/living/user, human_occupant = 0)
+	if(human_occupant)
+		eject_action.Grant(user, src)
+	internals_action.Grant(user, src)
+	cycle_action.Grant(user, src)
+	lights_action.Grant(user, src)
+	stats_action.Grant(user, src)
+	strafing_action.Grant(user, src)
+
+
+/obj/mecha/proc/RemoveActions(mob/living/user, human_occupant = 0)
+	if(human_occupant)
+		eject_action.Remove(user)
+	internals_action.Remove(user)
+	cycle_action.Remove(user)
+	lights_action.Remove(user)
+	stats_action.Remove(user)
+	strafing_action.Remove(user)
+
+
+/datum/action/innate/mecha
+	check_flags = AB_CHECK_RESTRAINED | AB_CHECK_STUNNED | AB_CHECK_CONSCIOUS
+	var/obj/mecha/chassis
+
+/datum/action/innate/mecha/Grant(mob/living/L, obj/mecha/M)
+	chassis = M
+	..()
+
+/datum/action/innate/mecha/Destroy()
+	chassis = null
+	return ..()
+
+/datum/action/innate/mecha/mech_eject
+	name = "Eject From Mech"
+	button_icon_state = "mech_eject"
+
+/datum/action/innate/mecha/mech_eject/Activate()
+	if(!owner || !iscarbon(owner))
+		return
+	if(!chassis || chassis.occupant != owner)
+		return
+	chassis.go_out()
+
+
+/datum/action/innate/mecha/mech_toggle_internals
+	name = "Toggle Internal Airtank Usage"
+	button_icon_state = "mech_internals_off"
+
+/datum/action/innate/mecha/mech_toggle_internals/Activate()
+	if(!owner || !chassis || chassis.occupant != owner)
+		return
+	chassis.use_internal_tank = !chassis.use_internal_tank
+	button_icon_state = "mech_internals_[chassis.use_internal_tank ? "on" : "off"]"
+	chassis.occupant_message("Now taking air from [chassis.use_internal_tank?"internal airtank":"environment"].")
+	chassis.log_message("Now taking air from [chassis.use_internal_tank?"internal airtank":"environment"].")
+	UpdateButtonIcon()
+
+/datum/action/innate/mecha/mech_cycle_equip
+	name = "Cycle Equipment"
+	button_icon_state = "mech_cycle_equip_off"
+
+/datum/action/innate/mecha/mech_cycle_equip/Activate()
+	if(!owner || !chassis || chassis.occupant != owner)
+		return
+
+	var/list/available_equipment = list()
+	for(var/obj/item/mecha_parts/mecha_equipment/M in chassis.equipment)
```

Comment:
Why itypeless on such a small and infrequently called code segment? Aren't unobvious microoptimizations like that reserved for really time-critical parts?